### PR TITLE
DEV-3515-feat(piscine-go): add solution for new go exercise stringtointarray

### PR DESCRIPTION
**Pull request by tiagocollot**
Created and added the solution for the new go exercise stringtointarray.

### CODE REVIEW

**Review by jotapero — APPROVED** [review]
Everything looks good

File: solutions/stringtointarray.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package solutions
+
+func StringToIntArray(str string) []int {
```
Comment (eslopfer) [inline solutions/stringtointarray.go:3]: As mentioned in the readme review, the name is misleading, it is a slice not an array.

File: tests/stringtointarray_test/main.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+func main() {
+	table := []string{"hello THERE" ,"Converted this string into an int", "14/%37", "a b c d e f g h i j k l m n o p q r s t u v w x y z", "a b c d e fghijk l m n o p q r s t u v w x y z ", "A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z", "heLLo There", "  121314", "qwçiodhqwd", " ", ""}
+
+	for _, s := range table {
+		challenge.Function("StringToIntArray", student.StringToIntArray, solutions.StringToIntArray, s)
```
Comment (eslopfer) [inline tests/stringtointarray_test/main.go:14]: Same as above

**Review by eslopfer — CHANGES_REQUESTED** [review]

File: solutions/stringtointarray.go (hunk 1 above)
Comment (tiagocollot) [reply solutions/stringtointarray.go]: It is fixed.

File: tests/stringtointarray_test/main.go (hunk 2 above)
Comment (tiagocollot) [reply tests/stringtointarray_test/main.go]: It is fixed.

**Review by eslopfer — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**